dts/bindings/mtd/st,m24c04-a125.yaml

```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2024 SILA Embedded Solutions GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+description: ST M24C04-A125 automotive 4-kbit I2C EEPROM
+
+compatible: "st,m24c04-a125"
+
+include: ["atmel,at2x-base.yaml", i2c-device.yaml]
+
+properties:
+  size:
+    const: 512
+  pagesize:
+    const: 16
+  address-width:
+    const: 8
```

Comment:
Do we actually need to have all these "non-compatible" compatibles using fixed properties?

What about a more generic `st,m24xxx` compatible ?
Aim it to document that a class of devices is supported and show how to configure it.

Reply:
I would rather prefer to be specific, as this has multiple advantages:

1. If I only document the proper usage, I still have to add some text which has to be maintained. And usually documentation gets forgotten during updates and changes and it is therefore more likely be incorrect in the future.
2. This adds a build time check, which avoids improper use and therefore frustration and bugs in real applications. And as the hardware won't change anymore I think this is worth the effort.
3. Having a very generic compatible brought us to the current situation. From my experience it is rather handy on the first sight to add a more generic compatible, but in the end hardware vendors tend to do all kind of weird stuff with their hardware identifiers. There might not be something currently in the range of st,m24xxx which is incompatible with this driver, but I wouldn't bet on this to be true in the future as well.
4. It is more clear to a user which compatible should be used for which IC in the hardware design.